### DISCUSSION TITLE
2.3.21

### DISCUSSION THREAD

**Discussion by benhillis** [microsoft/WSL#12037] [General]
*  Don't register the binfmt_late entry when interop is disabled (solves #11920)
*  Fix invalid iterator access causing service crash in DnsResolver::Stop
*  Simplify elevated relaunch to use AttachConsole instead of pseudoconsole
*  Update wsl.exe --mount to allow ARM64 passthrough disks on version of Windows that support it (solves #9611)
*  Update MSRDC to 1.2.5620
*  Update WinUIEx version 2.4.0
*  Update Microsoft.Windows.SDK.NET.Ref version 10.0.26100.45
*  Update Microsoft.Windows.ImplementationLibrary to version 1.0.240822.1
*  Update CommunityToolkit.Common and CommunityToolkit.Mvvm to version 8.3.1
*  Update various Microsoft runtime nuget packages from 8.0.7 -> 8.0.8

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/WSL/releases/tag/2.3.21'>2.3.21</a>.</em>

**Comment by mwoodpatrick** [comment]
This install is currently hanging for me, anyone else having any issues.

finally failed with:

Update failed (exit code: 1618).
Log file: C:\Users\mlwp\AppData\Local\Temp\wsl-install-logs.txt
Error code: Wsl/UpdatePackage/0x80070652



The log was empty

**Comment by g2flyer** [comment]
For what's worth: Didn't hang for me but update seemed to take longer than usual ...

**Comment by DoctorWho8** [comment]
For myself the update was successful, but really a kernel downgrade. How come this release installed a 5.15 series kernel? What I run, was working fine on the 6 series kernel the last update installed. So what was this one supposed to fix with a 5.15 kernel? BTW I run Slackware Linux as packaged by a group of enthusiasts.

**Comment by zcobol** [comment]
@DoctorWho8 what version you upgraded from? `2.3.17` reverted the Linux kernel to `5.15.153.1-2` while issues are being debugged.

**Reply by DoctorWho8** [reply]
2.2.4 I believe. I saw no issues that @zcobol you yourself saw. In fact I stayed with the the 6 series kernel because I saw nothing at all going on that could be caused a problem.

**Comment by zcobol** [comment]
`2.2.4` upgraded to `2.3.21`, no issues:

```
PS C:\Users\zcobol> wsl --version
WSL version: 2.2.4.0
Kernel version: 5.15.153.1-2
WSLg version: 1.0.61
MSRDC version: 1.2.5326
Direct3D version: 1.611.1-81528511
DXCore version: 10.0.26091.1-240325-1447.ge-release
Windows version: 10.0.22631.4169

PS C:\Users\zcobol> wsl --update --pre-release
Checking for updates.
Updating Windows Subsystem for Linux to version: 2.3.21.

PS C:\Users\zcoblo> wsl --version
WSL version: 2.3.21.0
Kernel version: 5.15.153.1-2
WSLg version: 1.0.65
MSRDC version: 1.2.5620
Direct3D version: 1.611.1-81528511
DXCore version: 10.0.26100.1-240331-1435.ge-release
Windows version: 10.0.22631.4169

```

**Comment by muratyurdakul75** [comment]
I eagerly look for dual numa support with every new version. But unfortunately it is not there. :( This is required in server systems. One of the 2 processors is wasted and there is unnecessary performance loss.

**Comment by mwoodpatrick** [comment]
I resolved my install issue, there was a dialog box that had gotten hidden, approving that fixed my install issue.
